pkg: go vet issues due to Seek signature

Hello! `io.Seeker` is one of the more core Go interfaces that should be followed for any method named `Seek`. For examle, go vet currently returns:

```
$ go vet ./...
# github.com/thanos-io/thanos/pkg/compact/downsample
pkg/compact/downsample/downsample.go:620:34: method Seek(x int64) bool should have signature Seek(int64, int) (int64, error)
pkg/compact/downsample/downsample_test.go:495:27: method Seek(int64) bool should have signature Seek(int64, int) (int64, error)
# github.com/thanos-io/thanos/pkg/query
pkg/query/iter.go:221:26: method Seek(int64) bool should have signature Seek(int64, int) (int64, error)
pkg/query/iter.go:237:34: method Seek(t int64) (ok bool) should have signature Seek(int64, int) (int64, error)
pkg/query/iter.go:288:32: method Seek(t int64) (ok bool) should have signature Seek(int64, int) (int64, error)
pkg/query/iter.go:519:32: method Seek(t int64) bool should have signature Seek(int64, int) (int64, error)
pkg/query/querier_test.go:688:26: method Seek(t int64) bool should have signature Seek(int64, int) (int64, error)
```

I don't know what your API compatibility rules and such are, but I would encourage giving this Seek either a different name or following the io.Seeker interface. Alternatively, you could consider whitelisting this and running vet in CI, but I would encourage complying with it to avoid unexpected surprises.

--- a/Makefile
+++ b/Makefile
@@ -298,6 +298,8 @@ web: web-pre-process $(HUGO)
 #
 # to debug big allocations during linting.
 lint: check-git $(GOLANGCILINT) $(MISSPELL)
+	@echo ">> examining all of the Go files"
+	@go vet -stdmethods=false
 	@echo ">> linting all of the Go files GOGC=${GOGC}"
 	@$(GOLANGCILINT) run
 	@echo ">> detecting misspells"

--- a/Makefile
+++ b/Makefile
@@ -299,7 +299,7 @@ web: web-pre-process $(HUGO)
 # to debug big allocations during linting.
 lint: check-git $(GOLANGCILINT) $(MISSPELL)
 	@echo ">> examining all of the Go files"
-	@go vet -stdmethods=false
+	@go vet -stdmethods=false ./pkg/... ./cmd/... && go vet doc.go
 	@echo ">> linting all of the Go files GOGC=${GOGC}"
 	@$(GOLANGCILINT) run
 	@echo ">> detecting misspells"

